src/ImageManipulation.php

```diff
@@ -709,6 +711,33 @@ public function ThumbnailURL($width, $height)
         return $this->getIcon();
     }
 
+    /**
+     * Convert the file to another format if there's a registered converter that can handle it.
+     *
+     * @param string $toFormat The file extension you want to convert to - e.g. "webp".
+     * @param bool $returnNullOnFailure If true and the conversion fails, `null` is returned instead.
+     * This is useful if calling this method from a template, for example, where you don't want to be throwing
+     * exceptions for end users.
+     * @throws FileConversionException If the conversion fails and $returnNullOnFailure is false.
+     */
+    public function convert(string $toFormat, bool $returnNullOnFailure = false): ?AssetContainer
```

Comment:
I originally had `$returnOriginalOnFailure` instead of `$returnNullOnFailure`, and it would return `$this` if the conversion failed. But `null` is more consistent with how other methods in this file behave, and in many cases it would be more desirable to display *nothing* in the template instead of *the wrong thing*.

Reply:
We could optionally remove that parameter and just **always** return `null` on failure. I'm actually leaning more and more towards that option the more that I think about it.... but I'll leave it as is for now pending review.

The main benefit of removing the parameter is that it is more consistent with other methods here. We'd probably want to at least log the failure though if we did that.
The main benefit of keeping it is that silent failing is an explicit decision developers have to make, which makes it safer.

Reply:
I think remove the param and just return null

You have written this "This is useful if calling this method from a template, for example, where you don't want to be throwing for end users."

That sounds like the main scenario this method will be called in? In which case let's not force the developer to have to put in a param, which they don't have to add in for any other template methods.

Reply:
I don't like that personally but I don't feel strongly enough about it to have peer review ping pong about it.

> That sounds like the main scenario this method will be called in?

There are all PHP methods and can be called from PHP. Any time you're dealing with an image (or file in this case) in PHP for any reason, you'll call these methods. File conversion will I suspect be just as common in PHP as it will in templates, for things like converting files before attaching to emails, etc.

That said, I've made the requested change and logged the error as a compromise.